### ISSUE TITLE
Treeview is not accessible using VoiceOver

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6666]
The is an issue collecting known problems using the VoiceOver (VO) screen reader on the experimental Treeview component.

## Environment

VoiceOver on Safari in latest macOS

## Issues:

- [ ] VO incorrectly describes the Treeview as a table made up of "listboxes" telling the user inccorect and non-functioning controls (note: testing against [the APG example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-1/treeview-1a.html) gave me similar results perhaps suggesting a browser/reader bug?)
  _VO should describe the Treeview as a list with disclosures (open/closed) for list items with nested nodes_

- [ ] Hitting enter or space bar selects the focused list item and does not toggle open/close when nested lists are present
  _The expected on click event for a tree view node with a sub-node is to toggle open/close state on `enter` or `space`_

- [ ] VO describes the SVG buttons as "Button, 1 of 1" even if it's the second, third, etc button focused in the list
  _Ideally we don't have nested SVG buttons being interacted with at all, but they should at least respect their count relative to each other_

- [ ] SVG buttons are not operable using `ctrl-shift+spacebar`
  _`ctrl-shift+spacebar` should 'click' the button opening or closing lists with nested lists in them_

**Comment by dakahn** [comment]
closing in favor of #6792
